app/javascript/src/PaymentGateways/components/StripeFormWrapper.jsx

```diff
@@ -0,0 +1,142 @@
+// @flow
+
+import React, { useState } from 'react'
+import { loadStripe } from '@stripe/stripe-js'
+import { Elements, CardElement, useStripe, useElements } from '@stripe/react-stripe-js'
+import { createReactWrapper } from 'utilities/createReactWrapper'
+import { CSRFToken } from 'utilities/utils'
+import 'PaymentGateways/styles/stripe.scss'
+
+type Props = {
+  stripePublishableKey: string,
+  setupIntentSecret: string,
+  billingAddressDetails: {
+    line1: string,
+    line2: string,
+    city: string,
+    state: string,
+    postal_code: string,
+    country: string
+  },
+  successUrl: string,
+  isCreditCardStored: boolean
+}
+
+const EditCreditCardDetails = ({isCreditCardStored}: {isCreditCardStored: boolean}) => {
+  const [isStripeFormVisible, setIsStripeFormVisible] = useState({ isStripeFormVisible: !isCreditCardStored })
+
+  const toggleStripeForm = () => {
+    const newStateStripeFormVisible = !isStripeFormVisible
+    const stripeForm = document.getElementById('stripe-form')
+    newStateStripeFormVisible ? stripeForm.classList.remove('hidden') : stripeForm.classList.add('hidden')
+    setIsStripeFormVisible(newStateStripeFormVisible)
+  }
+
+  return (
+    <a className='card-on-file' onClick={toggleStripeForm}>
+      <i className='fa fa-pencil'></i>
+      <span>{isStripeFormVisible ? 'cancel' : 'Edit Credit Card Details'}</span>
+    </a>
+  )
+}
+
+const CreditCardErrors = ({cardErrorMessage}: {cardErrorMessage: ?string}) => {
+  return (
+    <span id='card-errors' role='alert'>
+      {cardErrorMessage}
+    </span>
+  )
+}
+
+const StripeElementsForm = ({ stripePublishableKey, setupIntentSecret, billingAddressDetails, successUrl, isCreditCardStored }: Props) => {
+  const stripePromise = loadStripe(stripePublishableKey)
+
+  const CheckoutForm = () => {
+    const stripe = useStripe()
+    const elements = useElements()
+
+    const [cardErrorMessage, setCardErrorMessage] = useState({ cardErrorMessage: null })
```

Comment:
Same here, you don't need an object, just the default value of `cardErrorMessage`